diff --git a/build/index.d.ts b/build/index.d.ts
index 575dc0e00c51ac7dac66c4e4885c2414ee98fa91..50717f586aba777a4239b726a90347d9c3dcea45 100644
--- a/build/index.d.ts
+++ b/build/index.d.ts
@@ -1,5 +1,8 @@
 import { JwtPayload } from 'jsonwebtoken';
+import { HttpHandler } from 'msw';
+
 declare const createJWKSMock: (jwksBase: string, jwksPath?: string) => {
+    handler: HttpHandler
     start: () => void;
     stop: () => void;
     kid: () => string;
diff --git a/build/index.js b/build/index.js
index 289a2dac6cd569eef5dbc9329914a7517b442829..bfe7abde0ad358d21b707cc2fa69f721b02b8955 100644
--- a/build/index.js
+++ b/build/index.js
@@ -1,13 +1,20 @@
 import { createJWKS, createKeyPair, signJwt } from './tools.js';
 import { setupServer } from 'msw/node';
-import { rest } from 'msw';
+import { http, HttpResponse } from 'msw';
+
 const createJWKSMock = (jwksBase, jwksPath = '/.well-known/jwks.json') => {
     const keypair = createKeyPair();
     const JWKS = createJWKS({
         ...keypair,
         jwksOrigin: jwksBase,
     });
-    const server = setupServer(rest.get(new URL(jwksPath, jwksBase).href, (_, res, ctx) => res(ctx.status(200), ctx.json(JWKS))));
+    
+    // we patch & expose the actual msw handler so we can plug it into our own server rather than
+    // creating a conflicting instance. we also bump the msw version to match our own to avoid any
+    // compatibility issues.
+    const handler = http.get(new URL(jwksPath, jwksBase).href, () => HttpResponse.json(JWKS));
+
+    const server = setupServer(handler);
     const kid = () => JWKS.keys[0].kid;
     const start = () => {
         server.listen({ onUnhandledRequest: 'bypass' });
@@ -17,6 +24,7 @@ const createJWKSMock = (jwksBase, jwksPath = '/.well-known/jwks.json') => {
     };
     const token = (token = {}) => signJwt(keypair.privateKey, token, kid());
     return {
+        handler,
         start,
         stop,
         kid,
diff --git a/package.json b/package.json
index e5a5c2fc6b6b5dfec0c88bbf2adf4a88ba33e18b..412355319913b9b7a3743974c47d29fb834c9a93 100644
--- a/package.json
+++ b/package.json
@@ -91,7 +91,7 @@
   "dependencies": {
     "base64-url": "^2.3.3",
     "jsonwebtoken": "^9.0.0",
-    "msw": "^1.2.2",
+    "msw": "2.2.14",
     "node-forge": "^1.3.1",
     "node-rsa": "^1.1.1"
   },
